litho-processor/src/main/java/com/facebook/litho/specmodels/generator/EventGenerator.java

```diff
@@ -119,16 +122,74 @@ static TypeSpecDataHolder generateEventDispatcher(EventDeclarationModel eventDec
             .addModifiers(Modifier.STATIC)
             .addParameter(ClassNames.EVENT_HANDLER, "_eventHandler");
 
-    eventDispatcherMethod.addStatement(
-        "final $T _eventState = new $T()", eventDeclaration.name, eventDeclaration.name);
+    final TypeElement eventRepresentedObject = (TypeElement) eventDeclaration.representedObject;
+    final boolean eventClassHasOnlyOneNonEmptyCtor =
+        TypeElementUtils.hasOnlyOneNonEmptyConstructor(eventRepresentedObject);
+
+    /*
+    * Check if our class has any non-empty constructors. If so we will instantiate the
+    * Event class using that one instead of accessing mutable fields.
+    * */
+    if (eventClassHasOnlyOneNonEmptyCtor) {
+      final StringBuilder ctorParamsPlaceholder = new StringBuilder();
+      final List<String> ctorParamNames = new ArrayList<>();
+
+      /*
+      * Iterate over the Event fields in order to achieve the following:
+      *
+      * 1) Add the parameters to the method under construction.
+      * 2) Create a parameter name placeholder for JavaPoet.
+      * 3) Keep a list of all parameter names.
+      * */
+      for (FieldModel fieldModel : eventDeclaration.fields) {
+        eventDispatcherMethod
+            .addParameter(fieldModel.field.type, fieldModel.field.name);
+
+        // We do not want ',' appended on the first loop
+        if (!ctorParamNames.isEmpty()) {
+          ctorParamsPlaceholder.append(", ");
+        }
+
+        ctorParamsPlaceholder.append("$L");
 
-    for (FieldModel fieldModel : eventDeclaration.fields) {
-      if (fieldModel.field.modifiers.contains(Modifier.FINAL)) {
-        continue;
+        ctorParamNames.add(fieldModel.field.name);
       }
+
+      /*
+      * Create a list with all the desired arguments to be for the following statement.
+      * These need to be the following:
+      *
+      * 1) Event class name (type declaration)
+      * 2) Event class name (class instantiation)
+      * 3) Event class constructor parameter names.
+      * */
+      final List<Object> ctorStatementArgs = new ArrayList<>();
+      ctorStatementArgs.add(0, eventDeclaration.name);
+      ctorStatementArgs.add(1, eventDeclaration.name);
```

Comment:
nit: Can omit index

Reply:
The addition of the index was used because I wanted to showcase that there is a reason for the duplication of that add operation, since it is the same value for the first two arguments in varargs below.

Reply:
True, totally agree with the intention. Sorry for not getting this right